models/db.py

```diff
@@ -1,9 +1,36 @@
-from gluon.tools import Auth
 db = DAL('sqlite://storage.sqlite',pool_size=1,check_reserved=['all'])
+from gluon.tools import Auth, Service, PluginManager
+
 auth = Auth(db)
 
+## This allows me to still be able to use the web2py auth module while still customising properly the fields
+db.define_table('auth_user',
+                Field('first_name', length=128, default=''),
+                Field('last_name', length=128, default=''),
+                Field('voters_registration_number',),
+                Field('email', length=128, default='', unique=True),
+                Field('password', 'password', length=512,readable=False, label='Password'),
+                Field('registration_key', length=512, writable=False, readable=False, default=''),
+                Field('reset_password_key', length=512, writable=False, readable=False, default=''),
+                Field('registration_id', length=512, writable=False, readable=False, default=''))
+
+#validators
+custom_auth_table = db['auth_user'] # get the custom_auth_table
+custom_auth_table.first_name.requires = IS_NOT_EMPTY(error_message=auth.messages.is_empty)
+custom_auth_table.last_name.requires = IS_NOT_EMPTY(error_message=auth.messages.is_empty)
+custom_auth_table.password.requires = [IS_STRONG(special=0,upper=0,min=6), CRYPT()]#CRYPT validtor here hadles salting of the pasword
+custom_auth_table.email.requires = [
+  IS_EMAIL(error_message=auth.messages.invalid_email),
+  IS_NOT_EMPTY(),
+  IS_NOT_IN_DB(db, custom_auth_table.email)]
+custom_auth_table.voters_registration_number.requires = [
+    IS_NOT_EMPTY(),
+    IS_NOT_IN_DB(db, custom_auth_table.voters_registration_number)
+]
+
+auth.settings.table_user = custom_auth_table # tell auth to use custom_auth_table
+
+auth.settings.login_next = URL('confirm') #This helps to change the redirected url on user registration or login
+
 
-auth.settings.extra_fields['auth_user']= [
-    Field('Voters Identification Number', 'string')
-    ]
-auth.define_tables(username=True)
\ No newline at end of file
+auth.define_tables(signature=False)
```

Comment:
From what I see at http://web2py.com/books/default/chapter/29/09/access-control, the "Setting signature=True adds user and date stamping to auth tables, to track modifications.", which is something good for auditing (for example, if a user's account is stolen and the password modified, we could see when that was modified). Why did you disable that?